src/sources/socket/tcp.rs

```diff
@@ -14,21 +14,56 @@ use crate::{
     tls::TlsEnableableConfig,
 };
 
-#[derive(Deserialize, Serialize, Debug, Clone)]
+/// TCP configuration for the `socket` source.
+#[configurable_component]
+#[derive(Clone, Debug)]
 pub struct TcpConfig {
+    /// The address to listen for connections on.
     address: SocketListenAddr,
+
+    #[configurable(derived)]
     keepalive: Option<TcpKeepaliveConfig>,
+
+    /// The maximum buffer size, in bytes, of incoming messages.
+    ///
+    /// Messages larger than this are truncated.
     max_length: Option<usize>,
+
+    /// The timeout, in seconds, before a connection is forcefully closed during shutdown.
     #[serde(default = "default_shutdown_timeout_secs")]
     shutdown_timeout_secs: u64,
+
+    /// Overrides the name of the log field used to add the peer host to each event.
+    ///
+    /// The value will be the peer host's address, including the port i.e. `1.2.3.4:9000`.
+    ///
+    /// By default, the [global `host_key` option](\(urls.vector_configuration)/global-options#log_schema.host_key) is used.
```

Comment:
Just noting that I think we'll want to replace these references to urls defined in the CUE docs.

Reply:
Yes, agreed. I had that thought as I was doing full-on Markdown links that I might as well just link to the existing Cue docs via Markdown.

I'll get that fixed up.